### PR TITLE
Feature/weather

### PR DESCRIPTION
**Pull request by ZelikSV**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,7 +38,7 @@
     "build": "react-scripts build",
     "test": "react-scripts test",
     "eject": "react-scripts eject",
-    "eslint": "eslint src/"
+    "eslint": "eslint src/ --fix"
```
Comment (motifej) [inline package.json]: it is has to be a new script! with name 

> fix-lint

File: src/actions/weather.actions.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+} from './types';
+
+export const loadWeather = value => ({
+  type: LOAD_WEATHER,
+  payload: value,
+});
+
+export const loadWeatherRequest = () => ({
+  type: LOAD_WEATHER_REQUEST,
+  onLoading: true,
```
Comment (motifej) [inline src/actions/weather.actions.js]: onLoading: true - move to a reducer

File: src/actions/weather.actions.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+});
+
+export const loadWeatherSuccess = data => ({
+  type: LOAD_WEATHER_SUCCESS,
+  payload: data,
+  onLoading: false,
```
Comment (motifej) [inline src/actions/weather.actions.js]: move to a reducer

File: src/components/SearchInputGoogle/index.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import React, { Component } from 'react';
+import { compose, withProps } from 'recompose';
+import { withScriptjs } from 'react-google-maps';
+import { StandaloneSearchBox } from 'react-google-maps/lib/components/places/StandaloneSearchBox';
+import apiKeyGoogle from './config';
+
+class PlacesWithStandaloneSearchBox extends Component {
+  componentWillMount() {
```
Comment (motifej) [inline src/components/SearchInputGoogle/index.js]: componentWillMount- deprecated

File: src/components/SearchInputGoogle/index.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    const refs = {};
+    this.setState({
+      onSearchBoxMounted: ref => {
+        refs.searchBox = ref;
```
Comment (motifej) [inline src/components/SearchInputGoogle/index.js]: this is bad solution to work with refs

File: src/components/SearchInputGoogle/index.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    return (
+      <>
+        <StandaloneSearchBox
+          ref={onSearchBoxMounted}
+          bounds={bounds}
+          onPlacesChanged={onPlacesChanged}>
+          <input
+            type="text"
+            placeholder="Enter your location"
+            style={{
```
Comment (motifej) [inline src/components/SearchInputGoogle/index.js]: this is a bad practice put style to render, try to use clases

File: src/components/SearchInputGoogle/index.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+          />
+        </StandaloneSearchBox>
+      </>
+    );
+  }
+}
+
+export default compose(
+  withProps({
+    googleMapURL: `https://maps.googleapis.com/maps/api/js?key=${apiKeyGoogle}&callback&libraries=places,geometry`,
```
Comment (motifej) [inline src/components/SearchInputGoogle/index.js]: no harcoded data in component

File: src/components/SearchInputGoogle/index.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    loadingElement: <div style={{ height: '100%' }} />,
```
Comment (motifej) [inline src/components/SearchInputGoogle/index.js]: no inline styles

File: src/components/SearchInputGoogle/index.js [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    containerElement: <div style={{ height: '400px' }} />,
+  }),
+  withScriptjs,
```
Comment (motifej) [inline src/components/SearchInputGoogle/index.js]: i am not sure that you need `withScriptjs `

File: src/components/Weather/index.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+        <div className={classes.weatherContent}>
+          <div className={classes.weatherContentTemp}>
+            <i>{weather.temp}</i>
+            <span>
+              <sup> o</sup>C
+            </span>
+          </div>
+          <img
+            className={classes.weatherContentImg}
+            src={`https://www.weatherbit.io/static/img/icons/${
```
Comment (motifej) [inline src/components/Weather/index.js]: no hardcoded data in component

File: src/reducers/weather.reducer.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
+import combineEvents from 'utils/combineEvents';
+import {
+  LOAD_WEATHER_REQUEST,
+  LOAD_WEATHER_SUCCESS,
+  LOAD_WEATHER_ERROR,
+} from 'actions/types';
+
+const initialState = {
+  weatherData: {
+    city_name: 'New York',
```
Comment (motifej) [inline src/reducers/weather.reducer.js]: we are in New York?

File: yarn.lock [hunk 12]
```diff
@@ -2011,7 +2011,7 @@ babel-register@^6.26.0:
     mkdirp "^0.5.1"
     source-map-support "^0.4.15"
 
-babel-runtime@^6.22.0, babel-runtime@^6.26.0:
+babel-runtime@^6.11.6, babel-runtime@^6.22.0, babel-runtime@^6.26.0:
```
Comment (motifej) [inline yarn.lock]: add this file to ignore, remove from commit

File: package.json (hunk 1 above)
Comment (ZelikSV) [reply package.json]: sorry, it's my fall. i forgot to remove it

File: src/pages/Weather/index.js [hunk 13]
```diff
@@ -0,0 +1,33 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import Weather from 'components/Weather';
+import { loadWeather } from 'actions/weather.actions';
+
+class WeatherWrap extends Component {
+  componentDidMount() {
+    const { loadWeatherAction } = this.props;
+    loadWeatherAction({ lat: 48.46, lng: 35.046 });
```
Comment (motifej) [inline src/pages/Weather/index.js]: what is this? where it is?

File: src/components/SearchInputGoogle/index.js [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+export default compose(
+  withProps({
+    googleMapURL: `${GOOGLE_MAP_URL}?key=${apiKeyGoogle}&callback&libraries=places,geometry`,
+    loadingElement: <div />,
+    containerElement: <div />,
+  }),
+  withStyles(styles),
+  withScriptjs,
```
Comment (motifej) [inline src/components/SearchInputGoogle/index.js]: remove

**Pull request merged**